src/main/java/org/prebid/server/bidder/medianet/MedianetBidder.java

```diff
@@ -0,0 +1,37 @@
+package org.prebid.server.bidder.medianet;
+
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.OpenrtbBidder;
+import org.prebid.server.json.JacksonMapper;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Medianet {@link Bidder} implementation.
+ */
+public class MedianetBidder extends OpenrtbBidder<Void> {
+
+    private static final Logger logger = LoggerFactory.getLogger(MedianetBidder.class);
+    private static final String EXTERNAL_URL_MACRO = "{{PREBID_SERVER_ENDPOINT}}";
+
+    public MedianetBidder(String endpointUrl, JacksonMapper mapper, String externalUrl) {
+        super(getEndpoint(endpointUrl, externalUrl),
+                RequestCreationStrategy.SINGLE_REQUEST,
+                Void.class,
+                mapper);
+    }
+
+    private static String getEndpoint(String originalEndpoint, String replacement) {
+        try {
+            replacement = URLEncoder.encode(replacement, StandardCharsets.UTF_8.displayName());
+        } catch (UnsupportedEncodingException e) {
+            logger.error("Unable to replace params in medianet endpoint");
+        }
+        return StringUtils.replace(originalEndpoint, EXTERNAL_URL_MACRO, replacement);
+    }
```

Comment:
As I see, replacement will be only one time, on bidder instance creation, so pls update `{{PREBID_SERVER_ENDPOINT}}` with your value, we don't need to do it in bidder, also lines 18 and 19 can be removed

Reply:
Or if everything as it should be, update code to 
```java
private static String getEndpoint(String endpoint, String replacement) {

        return endpoint.replace(EXTERNAL_URL_MACRO, HttpUtil.encodeUrl(replacement));
    }
```

Reply:
We have passed the modified url in MedianetConfiguration. We are using HttpUtil now.